player/lua/auto_profiles.lua

```diff
@@ -14,6 +14,9 @@ local pending_hooks = {}            -- as set (keys only, meaningless values)
 -- profile the condition is evaluated for.
 local current_profile = nil
 
+-- Cached list of all top-level mpv properities. Only used for extra validation.
+local property_list = mp.get_property_native("property-list")
```

Comment:
Turning that into a set would allow you to check for the existence of properties without iterating over it `property_set[name]`

Reply:
Yeah duh, use a better data structure. Not sure if there's a more elegant way to turn that table to a list in lua but this works.